backend/de.metas.business/src/main/java/de/metas/order/model/interceptor/C_OrderLine.java

```diff
@@ -445,4 +457,14 @@ public void updateTaxFromVatCodeId(final I_C_OrderLine orderLine)
 			orderLine.setC_Tax_ID(tax.getTaxId().getRepoId());
 		}
 	}
+
+
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_NEW, ModelValidator.TYPE_AFTER_CHANGE }, //
+			ifColumnsChanged = { I_C_OrderLine.COLUMNNAME_M_AttributeSetInstance_ID })
+	public void updateIsOnConsignment(final I_C_OrderLine orderLine)
+	{
+		final I_C_Order order = orderLine.getC_Order();
+		orderBL.setIsOnConsignmentOrder(order, orderBL.isOnConsignmentOrder(order));
```

Comment:
that's a clear sign that u don't need all those methods exposed in the OrderBL

instead, only one method shall be sufficient

IOrderBL.updateIsConsignmentFromLines(OrderId orderId)

----
for the records:

> orderLine.getC_Order();

pls avoid model getters, use DAO.